src/api/reflector.rs

```diff
@@ -211,7 +213,13 @@ impl<K> Reflector<K> where
         let rg = &self.resource;
         let oldver = { self.version.read().unwrap().clone() };
         let req = rg.watch(&self.params, &oldver)?;
-        let res = self.client.request_events::<WatchEvent<K>>(req).await?;
+        let res = self
+            .client
+            .request_events::<WatchEvent<K>>(req)
+            .await?
+            .filter_map(|e| async move { e.ok() })
+            .collect::<Vec<_>>()
+            .await;
```

Comment:
Reflector doing the full wait internally is totally fine, but it might make sense to take the lock, and update each object as they appear. Again, that is a more ambitious change.

Reply:
Makes sense, would this mean changing `Reflector::Poll` to return a `Stream<Item = Result<()>>`? or would you want it to return a `Stream<Item = Result<K>>` and then similarly add a stream handling step that updates the reflector's state based on the event and then return the K back?

Reply:
Looking at [the pod reflector example](https://github.com/clux/kube-rs/blob/7dab94601f2fb42d1f047037343f2b70350fa341/examples/pod_reflector.rs#L30), I'm a bit confused on what it would look like if `poll` returned a `Stream`

Reply:
The reflector shouldn't really return a stream in any way, because it acts as an on-demand cache.

we might need to change the names of the methods in one of Reflector/Informer for async.

Reply:
It should probably still do the book-keeping internally when the events come in though!

Reply:
Ahh got it 👍 , okay I've updated the reflect `single_watch` then to use the stream in it's handling instead of collecting into an intermediate `Vec` but still drain the stream entirely, so a call to `poll` will result in the reflector being up-to-date